test_server.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+
+import pytest
+import socket
+from http_server import response_ok, response_error
+
+
+@pytest.fixture()
+def make_client():
+    ADDR = ("127.0.0.1", 8010)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.connect(ADDR)
+    return client
+
+
+def helper(client, msg):
+    try:
+        client.sendall(msg)
+        client.shutdown(socket.SHUT_WR)
+        response = ""
+        while True:
+            part = client.recv(16)
+            response += part
+            if len(part) < 16:
+                client.close()
+                return response
+    except Exception as e:
+        print e
+
+
+def test_client1(make_client):
+    client = make_client
+    assert helper(client, "What I put here does not matter") == "HTTP/1.1 200 OK <html><head><title>Success!</title></head><body><header><h1>Success!</h1></header><p>It worked</p></body></html>"
+
+
+def test_response_ok():
+    assert "HTTP/1.1 200 OK" in response_ok()
```

Comment:
This test is certainly necessary but is not sufficient.  There's quite a bit more that goes into making a bytestring a proper, well-formed HTTP response.  Your tests should cover those bases.